### PR TITLE
sqlutils: narrow the testing shim

### PR DESCRIPTION
**Pull request by tbg**
The interface used to take `*testing.T` could be trimmed down a bit,
which helps with https://github.com/cockroachdb/cockroach/pull/81516.

Release note: None

### CODE REVIEW

**Review by mgartner — APPROVED** [review]
LGTM

File: pkg/testutils/sqlutils/sql_runner.go [hunk 1]
```diff
@@ -233,11 +261,11 @@ func MatrixToStr(rows [][]string) string {
...
 // CheckQueryResults checks that the rows returned by a query match the expected
 // response.
-func (sr *SQLRunner) CheckQueryResults(t testutils.TB, query string, expected [][]string) {
-	t.Helper()
+func (sr *SQLRunner) CheckQueryResults(t Fataler, query string, expected [][]string) {
+	helperOrNoop(t)()
 	res := sr.QueryStr(t, query)
 	if !reflect.DeepEqual(res, expected) {
-		t.Errorf("query '%s': expected:\n%v\ngot:\n%v\n",
+		t.Fatalf("query '%s': expected:\n%v\ngot:\n%v\n",
```
Comment (srosenberg) [inline pkg/testutils/sqlutils/sql_runner.go:268]: Are these really equivalent in this context? `Errorf` invokes `Fail` whereas `Fatalf` invokes `FailNow` (in the stdlib `testing` package).

Comment (tbg) [reply pkg/testutils/sqlutils/sql_runner.go:268]: They're not equivalent, but there is little justification for why Errorf was a better choice initially. We usually prefer a `require`-style over an `assert`-style by default when a `t` is passed to a function.

Comment (srosenberg) [reply pkg/testutils/sqlutils/sql_runner.go:268]: > They're not equivalent, but there is little justification for why Errorf was a better choice initially. We usually prefer a `require`-style over an `assert`-style by default when a `t` is passed to a function.

Got it, thanks! I am a little peeved that every language has its own semantics for `require` and `assert`; in static verification, they have the same semantics (the former is essentially translated to the latter). 

FYI, Google's popular lib. for java (guava) throws an exception for both (`IllegalArgumentException` vs `IlegalStateException`); scala's `require` throws `IllegalArgumentException` while `assert` throws `AssertionError`. Effectively, every PL/framework prefers to impose its own style.

**Pull request merged**